include/BSplineX/knots/t_atter.hpp

```diff
@@ -169,7 +169,10 @@ class Atter
       return retval;
     }
 
-    difference_type operator-(iterator const &b) const { return this->index - b.index; }
+    difference_type operator-(iterator const &b) const
+    {
+      return static_cast<int>(this->index - b.index);
```

Comment:
Should this be casted to `difference_type`?
```suggestion
      return static_cast<difference_type>(this->index - b.index);
```

Reply:
Yes

Reply:
Not related, but we should probably change `difference_type` from `int` to `std::ptrdiff_t`